### PR TITLE
[Metadata export]Add external features in analyzer

### PR DESCRIPTION
**Pull request by yuli-han**
## Summary

Add external features as enriched feature in analyzer to be exported to UMS

## Test Plan

- [x] Added Unit Tests
- [ ] Covered by existing CI
- [x] Integration tested

Added external feature: `{"window":"","name":"ext_contextual_search_request_id","columnType":"string","operation":"","groupBy":"contextual_values","inputColumn":"search_request_id"}`

## Checklist
- [ ] Documentation update

## Reviewers
@pengyu-hou @hzding621 @SophieYu41

### CODE REVIEW

File: spark/src/main/scala/ai/chronon/spark/Analyzer.scala [hunk 1]
```diff
@@ -310,6 +310,18 @@ class Analyzer(tableUtils: TableUtils,
       if (gbStartPartition.nonEmpty)
         gbStartPartitions += (part.groupBy.metaData.name -> gbStartPartition)
     }
+    if (joinConf.onlineExternalParts != null) {
+      joinConf.onlineExternalParts.toScala.foreach { part =>
+        aggregationsMetadata ++= part.source.valueSchema.params.toScala.map { param =>
+          AggregationMetadata(part.fullName + "_" + param.name,
+                              DataType.fromTDataType(param.dataType),
+                              "",
+                              "",
```
Comment (pengyu-hou) [inline spark/src/main/scala/ai/chronon/spark/Analyzer.scala:319]: should we leave them as null?

**Review by pengyu-hou — APPROVED** [review]
LGTM with one minor comments.

**Review by hzding621 — COMMENTED** [review]
could you update the unit test?

File: spark/src/main/scala/ai/chronon/spark/Analyzer.scala [hunk 2]
```diff
@@ -310,6 +310,18 @@ class Analyzer(tableUtils: TableUtils,
       if (gbStartPartition.nonEmpty)
         gbStartPartitions += (part.groupBy.metaData.name -> gbStartPartition)
     }
+    if (joinConf.onlineExternalParts != null) {
+      joinConf.onlineExternalParts.toScala.foreach { part =>
+        aggregationsMetadata ++= part.source.valueSchema.params.toScala.map { param =>
```
Comment (hzding621) [inline spark/src/main/scala/ai/chronon/spark/Analyzer.scala:315]: use part.source.valueFields

**Comment by hzding621** [comment]
Could you also update line 334 - 344 to include schema for external parts?

**Comment by yuli-han** [comment]
> Could you also update line 334 - 344 to include schema for external parts?

Hi @hzding621 The external features are included in the right schema.

File: spark/src/main/scala/ai/chronon/spark/Analyzer.scala (hunk 2 above)
Comment (yuli-han) [reply spark/src/main/scala/ai/chronon/spark/Analyzer.scala]: updated

File: spark/src/main/scala/ai/chronon/spark/Analyzer.scala (hunk 1 above)
Comment (yuli-han) [reply spark/src/main/scala/ai/chronon/spark/Analyzer.scala]: update

**Review by hzding621 — APPROVED** [review]

**Pull request merged**